invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -559,3 +559,32 @@ def show_anns(self, anns: List[Dict]):
             img[:, :] = ann_color
             final_img.paste(Image.fromarray(img, mode="RGB"), (0, 0), Image.fromarray(np.uint8(m * 255)))
         return np.array(final_img, dtype=np.uint8)
+
+
+@invocation(
+    "color_map_image_processor",
+    title="Color Map Processor",
+    tags=["controlnet"],
+    category="controlnet",
+    version="1.0.0",
+)
+class ColorMapImageProcessorInvocation(ImageProcessorInvocation):
+    """Generates a color map from the provided image"""
+
+    map_resolution: int = InputField(default=512, ge=0, description=FieldDescriptions.image_res)
+
+    def run_processor(self, image: Image.Image):
+        image = image.convert("RGB")
+        image = np.array(image, dtype=np.uint8)
+        height, width = image.shape[:2]
+
+        rescaled_width = int(round(width / height * self.map_resolution)) if height < width else self.map_resolution
+        rescaled_height = self.map_resolution if height < width else int(round(height / width * self.map_resolution))
+        rescaled_image = cv2.resize(image, (rescaled_width, rescaled_height), interpolation=cv2.INTER_AREA)
+
+        color_map = cv2.resize(
+            rescaled_image, (rescaled_width // 64, rescaled_height // 64), interpolation=cv2.INTER_CUBIC
```

Comment:
Is the hard-coded downscaling rate (`64`) based on some reference implementation? I think we could expose this as a parameter. I suspect that the important thing is just that this downscaling factor is large enough to remove the original image detail.

Reply:
64 seemed like a good spot because it gives us 16 px on each side to average the color. Lower values really don't do much because we get really close to looking and feeling like the original image (with the finer details) which defies the purpose of a color map. We can go higher for more dissipated color maps. Can easily expose this value.

I'll make a new PR for the color processor invocation, extract the resizer and expose the sampling region size.

One downside to this would be that the user will need to end up processing more nodes to achieve the same and we'll also end up saving another image to disk (which may not entirely be needed).